tools/killswitch/src/killswitch.rs

```diff
@@ -0,0 +1,258 @@
+        chain_setup
+            .try_into_connection_manager(Some(submitter_config.clone()), None)
+            .await
+            .map_err(|report| Error::ConnectionManagerInit(format!("{:#}", report)))
+    }
+
+    /// Make `Signers` or return error
+    async fn make_signer(channel: &Channel, settings: &Settings) -> Result<Signers> {
+        let config = settings
+            .attestation_signers
+            .get(&channel.home)
+            .ok_or_else(|| Error::MissingAttestationSignerConf(channel.home.clone()))?;
+        Signers::try_from_signer_conf(config)
+            .await
+            .map_err(|report| Error::AttestationSignerInit(format!("{:#}", report)))
+    }
+
+    /// Build a new `KillSwitch`, configuring best effort and storing, not returning most errors
+    pub(crate) async fn new(args: Args, settings: Settings) -> Result<Self> {
+        let channels = if args.all {
+            Self::make_channels(&settings)
+        } else {
+            let destination_network = args
+                .all_inbound
+                .expect("Should not happen. Clap requires this to be present");
+            let all = Self::make_channels(&settings);
+            Self::make_inbound_channels(&destination_network, all)
+        };
+        if channels.is_empty() {
+            // The one error we bail on, since there's nothing else left to do
+            return Err(Error::NoNetworks);
+        }
+
+        let futs = channels.into_iter().map(|channel| async {
+            let home_contract = Self::make_home(&channel, &settings).await;
+            let connection_manager = Self::make_connection_manager(&channel, &settings).await;
+            let attestation_signer = Self::make_signer(&channel, &settings).await;
+            ChannelKiller {
+                channel,
+                home_contract: Some(home_contract),
+                connection_manager: Some(connection_manager),
+                attestation_signer: Some(attestation_signer),
+            }
+        });
+        let channel_killers = join_all(futs).await.into_iter().collect::<Vec<_>>();
+
+        Ok(Self { channel_killers })
+    }
+
+    /// Collect all blocking errors, returning a `KillSwitch` with a set of channels
+    /// that can actually fire off transactions as well as any errors collected
+    pub(crate) async fn get_blocking_errors(self) -> (Self, Option<Message>) {
+        let (mut failed, maybe_ok): (Vec<_>, Vec<_>) = self
+            .channel_killers
+            .into_iter()
+            .partition(|killer| killer.has_errors());
+
+        let _blocking_errors = failed
+            .iter_mut()
+            .map(|killer| (killer.channel.clone(), killer.take_all_errors()))
+            .collect::<Vec<(_, _)>>();
+
+        // Produce errors to stream before running txs
+        let message = None;
+
+        (
+            KillSwitch {
+                channel_killers: maybe_ok,
+            },
+            message,
+        )
```

Comment:
Just a flag that we still haven't piped errors into set of returned messages in case this was forgotten